### PR TITLE
VTK io fixes

### PR DESCRIPTION
**Pull request by VictorLamoine**
- Return value is now a boolean instead of the number of faces in the model
- Enable user to choose between ASCII/binary formats

### CODE REVIEW

**Comment by taketwo** [comment]
Why do you think returning `bool` is better? `int` is anyway implicitly convertible to boolean, so I don't see any benefits in changing already existing interface. There might be people relying on it.

But nevertheless, your are correct that the output value of `poly_writer->Write ()` should be taken in the account.

**Comment by VictorLamoine** [comment]
In the case the writer didn't write all the faces (eg: full file system) the number of faces(different from 0) returned will be converted to a `true` boolean while the writing operation was not successful.

It's safer to check `poly_writer->Write ()`

File: io/src/vtk_lib_io.cpp [hunk 1]
```diff
@@ -74,26 +74,35 @@ pcl::io::loadPolygonFile (const std::string &file_name, pcl::PolygonMesh& mesh)
...
   // TODO: what about sensor position and orientation?!?!?!?
-  // TODO: how to adequately catch exceptions thrown by the vtk writers?!
   std::string extension = file_name.substr (file_name.find_last_of (".") + 1);
   if (extension == "pcd") // no Polygon, but only a point cloud
   {
-    int error_code = pcl::io::savePCDFile (file_name, mesh.cloud);
+    int error_code;
+    if (binary_format)
+    {
+      pcl::PointCloud<pcl::PointXYZRGB> cloud;
```
Comment (taketwo) [inline io/src/vtk_lib_io.cpp]: Hm, seems like my comment on this line disappeared somehow. Repeating: why this specific point type?

**Comment by taketwo** [comment]
Ok, agree.

File: io/src/vtk_lib_io.cpp (hunk 1 above)
Comment (VictorLamoine) [inline io/src/vtk_lib_io.cpp]: It has been lost [here](https://github.com/VictorLamoine/pcl/commit/faede2996010f268854c53e0bb5dc5eca528a9dc#commitcomment-12306413).

Why not? I don't know how I should handle this kind of case :sweat_smile:

Comment (taketwo) [inline io/src/vtk_lib_io.cpp]: What if the mesh cloud contained different channels than XYZRGBA?

But anyway, I'm not sure why you make this cast for the "binary" case. The [savePCDFile()](https://github.com/PointCloudLibrary/pcl/blob/master/io/include/pcl/io/pcd_io.h#L563) function allows to save blob point clouds in binary mode as well.

Comment (VictorLamoine) [inline io/src/vtk_lib_io.cpp]: That's right, I missed this one. Fixed

File: io/src/vtk_lib_io.cpp [hunk 2]
```diff
@@ -74,26 +74,35 @@ pcl::io::loadPolygonFile (const std::string &file_name, pcl::PolygonMesh& mesh)
...
+      error_code = pcl::io::savePCDFile (file_name, mesh.cloud, origin, orientation, true);
+    }
+    else
+      error_code = pcl::io::savePCDFile (file_name, mesh.cloud);
+
     if (error_code != 0)
-      return (0);
-    return (static_cast<int> (mesh.cloud.width * mesh.cloud.height));
+      return (false);
+    return (true);
```
Comment (taketwo) [inline io/src/vtk_lib_io.cpp]: What about collapsing these 15 lines of code (85-99) into something like:

```cpp
return (pcl::io::savePCDFile (file_name, mesh.cloud, Eigen::Vector4f::Zero (),
                              Eigen::Quaternionf::Identity (), binary_format) == 0);
```

Comment (VictorLamoine) [inline io/src/vtk_lib_io.cpp]: Oh yes :)

**Pull request merged**